jobsdb/jobsdb.go

```diff
@@ -1450,14 +1447,22 @@ func (jd *HandleT) constructParameterJSONQuery(table string, parameterFilters []
 	return fmt.Sprintf(`(%s.parameters @> '{%s}' %s)`, table, strings.Join(allKeyValues, ","), opQuery)
 }
 
+type cacheValue int
+
+const (
+	mayFindJobs cacheValue = iota
+	noJobs
+	willTryToSet
+)
+
 /*
 * If a query returns empty result for a specific dataset, we cache that so that
 * future queries don't have to hit the DB.
 * markClearEmptyResult() when mark=True marks dataset,customVal,state as empty.
 * markClearEmptyResult() when mark=False clears a previous empty mark
  */
 
-func (jd *HandleT) markClearEmptyResult(ds dataSetT, stateFilters []string, customValFilters []string, parameterFilters []ParameterFilterT, mark bool) {
+func (jd *HandleT) markClearEmptyResult(ds dataSetT, stateFilters []string, customValFilters []string, parameterFilters []ParameterFilterT, value cacheValue, checkAndSet *cacheValue) {
```

Comment:
Why make checkAndSet a pointer?

Reply:
Right now, cacheValue is of type int and I wanted to have nil support to disable CheckAndSet behaviour in the function.